packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx

```diff
@@ -406,6 +406,56 @@ export const DeckPlugin = ({
               };
             }
 
+            case NavigationAction.ADD_TO_ACTIVE: {
+              const { id, scrollIntoView, pivot } = intent.data as NavigationAction.AddToActive;
+              batch(() => {
+                if (isActiveParts(location.active)) {
+                  const main = Array.isArray(location.active.main) ? [...location.active.main] : [location.active.main];
+
+                  if (!main.includes(id)) {
+                    // Check if the id is not already in the main array
+                    if (pivot) {
+                      const pivotIndex = main.indexOf(pivot.id);
+                      if (pivotIndex !== -1) {
+                        main.splice(pivotIndex + (pivot.position === 'add-after' ? 1 : 0), 0, id);
+                      } else {
+                        main.push(id);
+                      }
+                    } else {
+                      const newIds = [id];
+                      const partMembers = new Set<string>();
+                      switch (settings.values.newPlankPositioning) {
+                        case 'start': {
+                          newIds.forEach((newId) => partMembers.add(newId));
+                          main.forEach((existingId) => partMembers.add(existingId));
```

Comment:
For this to work as expected, you have to filter `newIds` to just the ones not already in `main`, otherwise the ids move around. The implementation of `NavigationAction.OPEN` adjacent this one does that.

Reply:
I see this is an array with one item, and moving it is probably the point of this action, so never mind 😅